pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -300,18 +300,20 @@ func (f *Fit) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldOb
 	return framework.Queue, nil
 }
 
-// isResourceScaleDown checks whether the resource request of the modified pod is less than the original pod
-// for the resources requested by the pod we are trying to schedule.
-func (f *Fit) isResourceScaleDown(targetPod, originalOtherPod, modifiedOtherPod *v1.Pod) bool {
-	if modifiedOtherPod.Spec.NodeName == "" {
```

Comment:
Side question: What if the unscheduled Pod was assumed?

Doesn't that have the potential to free resources?

Reply:
I missed this scenario. we should also check if the modified pod is assumed when its nodename is empty.

Reply:
Add a new discussion here https://github.com/kubernetes/kubernetes/issues/122406, I think we can ignore the assumed pod for simple design.

Reply:
If a pod is being assumed, the hintFn won't observe update event because the event is filtered by the [updatePodInSchedulingQueue](https://github.com/kubernetes/kubernetes/blob/42ab3b7b40c34e91b63589ae905ed918752a4a36/pkg/scheduler/eventhandlers.go#L132). so we don't need consider this scenario

Reply:
> Side question: What if the unscheduled Pod was assumed?
https://github.com/kubernetes/kubernetes/pull/122360#discussion_r1431547953

I think @alculquicondor meant "unassumed" because assuming Pod doesn't make a free space, but just consume a space in some Node (as a new Pod will go there). Reversely "unassume" could make a free space.

1. Only one Node which can accomodate one Pod exists. 
2. Pod-a is assumed to the Node.
3. Pod-b is rejected because the scheduler assumes Pod-a.
4. Pod-a is somehow unassumed. 
5. Pod-a goes to the scheduling queue. Now, the Node has a free space though, Pod-b won't be retried.

@carlory Can you check if ^ this scenario could happen? If yes, we have to have some fix for this.

Reply:
(deleted one previous comment as it complicates the discussion here 😅 I'll create another issue for that.)

Reply:
Let's continue this discussion in @kerthcet 's issue.